### PR TITLE
Issue 249 - Node alias

### PR DESCRIPTION
**Pull request by emilianodellacasa**
For issue #249 I added a new _alias_ parameter to _node_ command to assign an alphanumeric string to a node. This string is then included in the mail REST responses.

Please note that _alias_ is a reserved word, so I hade to call the internal attribute _node_alias_ to avoid conflicts

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#437](https://github.com/zold-io/zold/issues/437) is now in scope, role is `REV`

**Comment by emilianodellacasa** [comment]
@yegor256 This PR is also still waiting for a REV, if you don't mind

**Comment by emilianodellacasa** [comment]
@yegor256 ping

File: lib/zold/commands/node.rb [hunk 1]
```diff
@@ -171,6 +174,13 @@ def run(args = [])
       Front.set(:dump_errors, opts['dump-errors'])
       Front.set(:port, opts['bind-port'])
       Front.set(:reboot, !opts['never-reboot'])
+      node_alias = opts[:alias] || address
+      unless node_alias.eql?(address)
+        re = Regexp.new(/^[a-z0-9]{4,16}$/)
+        puts re.match(node_alias)
```
Comment (yegor256) [inline lib/zold/commands/node.rb]: @emilianodellacasa typo?

File: lib/zold/node/front.rb [hunk 2]
```diff
@@ -66,6 +66,7 @@ class Front < Sinatra::Base
       set :wallets, nil? # to be injected at node.rb
       set :remotes, nil? # to be injected at node.rb
       set :copies, nil? # to be injected at node.rb
+      set :node_alias, nil? # to be injected at node.rb
```
Comment (yegor256) [inline lib/zold/node/front.rb]: @emilianodellacasa just `alias` is not descriptive enough? https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

**Comment by yegor256** [comment]
@emilianodellacasa see above

File: lib/zold/commands/node.rb (hunk 1 above)
Comment (emilianodellacasa) [reply lib/zold/commands/node.rb]: It's for validating the alias, why should I remove it?

File: lib/zold/node/front.rb (hunk 2 above)
Comment (emilianodellacasa) [reply lib/zold/node/front.rb]: @yegor256 The fact is that _alias_ is a reserved word for Ruby and it is not accepted, I tried again now and it gives a lot of syntax errors

**Comment by emilianodellacasa** [comment]
@yegor256 answered, see comments

**Comment by emilianodellacasa** [comment]
@yegor256 ping

File: lib/zold/commands/node.rb (hunk 1 above)
Comment (yegor256) [reply lib/zold/commands/node.rb]: @emilianodellacasa we don't use `puts`, we use logging instead

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#437](https://github.com/zold-io/zold/issues/437) is now out of scope

**Pull request merged**